### DISCUSSION TITLE
Chain validation not working in Android

### DISCUSSION THREAD

**Discussion by CICADA-dev64bit** [Kaliumhexacyanoferrat/GenHTTP#571] [Q&A]
Hey @Kaliumhexacyanoferrat, 

I came across a issue with Android device, the chain validation we have written that is not working in android but works perfectly fine in Windows. Do you think this issue is related to the GenHTTP server or the issue lies within our code?
Below i have attached the code for chain validation which functions in windows but not in android.

```csharp
using (var chainValidation = new X509Chain())
{
     chainValidation.ChainPolicy.ExtraStore.Add(embeddedCACertificate);
     chainValidation.ChainPolicy.RevocationMode = X509RevocationMode.NoCheck;
     chainValidation.ChainPolicy.VerificationFlags = X509VerificationFlags.AllowUnknownCertificateAuthority;
     bool isValid = chainValidation.Build(serverCertificate);	  
	 if(!isValid){
		 foreach (var status in chainValidation.ChainStatus)
		{
			Console.WriteLine($"Chain error: {status.StatusInformation}");
		}
		return false;
	 }
}
```

**Comment by Kaliumhexacyanoferrat** [comment]
To be honest I am no expert in client certificate validation. Just to be clear:

- This code is executed to validate the certificate of a connecting client
- This code is either executed on Android or Windows (so the server runs on Android as well?)

Or is the server always on Linux/Windows and the connecting client is either Windows (which works) or Anroid (which does not work)?

**Reply by Kaliumhexacyanoferrat** [reply]
@CICADA-dev64bit did you solve your issue? 😉

**Comment by Kaliumhexacyanoferrat** [comment]
As this has been stale for some time I will close the discussion - feel free to re-open it if there is still something to clarify.
